src/main/resources/Development-documentation/gerrit-scenarios.md

```diff
@@ -131,27 +159,84 @@ The user does the following:
 - Squashes the commits to one commits (`C1`)
 - Pushes to `refs/for/[branch name]` (`P1`)
 - Receives comments from reviewer
-- Updates code
+- Updates the code
 - Does `commit --amend` (`C2`)
-- Pushes to `refs/for/[branch name]`  (`P2`)
+- Pushes to `refs/for/[branch name]` (`P2`)
+- Gets ok from the reviewer
+- Hits the submit button in Gerrit
+- Gerrit will rebase (`C3`)
+
+Eiffel events sent from the plugin:
+
+- SCC(`E1`) sent for `P1` push with `BASE` link set to `E0`
+- SCC(`E2`) sent for `P2` push with `PREVIOUS_VERSION` link set `E1` and `BASE` link set to `E0`
+- SCS(`E3`) sent at submit with `CHANGE` link set `E2` and `PREVIOUS_VERSION` link set to `E01`
+
+Commit history after submit:
+
+```git
+* C3 Rebased changes
+* C01 Other code change
+* C0 Commit from previous user
+```
+
+Trigger table:
+
+| Event | Triggered by | Comment |
+| ---   | ---          | ---     |
+| E0    |              | Submit  |
+| E1    | P1           |
+| E2    | P2           |
+| E3    |              | Submit  |
+
+SCC Event table:
+
+| Event | BASE | PREVIOUS_VERSION |
+| ---   | ---  | ---              |
+| E1    | E0   |
+| E2    | E0   | E1               |
+
+## Create patch set, submitting with merge strategy - CASE4
+
+Scenario overview:
+
+This scenario describes a workflow where a user performs a standard review
+cycle. Due to project setting **Always merge** Gerrit will preform a merge when
+the user submits the review.
+
+Preconditions:
+
+- Previous review submitted to the master with hash `C0`
+- The plugin sends SCS with id `E0`
+
+The user does the following:
+
+- Creates a branch
+- Updates the code
+- Squashes the commits to one commits (`C1`)
+- Pushes to `refs/for/[branch name]` (`P1`)
+- Receives comments from reviewer
+- Updates the code
+- Does `commit --amend` (`C2`)
+- Pushes to `refs/for/[branch name]` (`P2`)
 - Gets ok from the reviewer
-- Forced to do a merge due to `C01` (`C3`) event `E01`
-- Pushes to `refs/for/[branch name]`  (`P3`)
 - Hits the submit button in Gerrit
+- Gerrit creates a merge commit (`C3`)
 
-Events sent:
+Eiffel events sent from the plugin:
 
 - SCC(`E1`) sent for `P1` push with `BASE` link set to `E0`
 - SCC(`E2`) sent for `P2` push with `PREVIOUS_VERSION` link set `E1` and `BASE` link set to `E0`
-- SCS(`E3`) sent for `P3` push with `PREVIOUS_VERSION` link set `E2` and `BASE` link set to `?`
-- SCS(`E4`) sent at submit with `CHANGE` link set `E3`
+- SCS(`E3`) sent at submit with `CHANGE` link set `E2`and `PREVIOUS_VERSION` set to `E0`
+
+**NOTE:** The plugin will not send a SCS event for `C2` and can not set a
+`PREVIOUS_VERSION` link to the commit.
```

Comment:
Of course it doesnt send an SCS for C2 as that commit is just a new patchset. Why should it set the previous version link for a patchset? It might just be so that I'm misunderstanding something 😄

Reply:
I also read this wrong at first. Perhaps clarify with: "The plugin will not send a SCS event for C2 when it's merged but instead for the C3 which is a merge commit made by Gerrit. This means there will be no PREVIOUS_VERSION link from C3 to C2"

Reply:
Fixed in 187495d

Reply:
I'm happy with that